Tests/SwiftParserTest/translated/IfconfigExprTests.swift

```diff
@@ -427,6 +427,26 @@ final class IfconfigExprTests: XCTestCase {
       ]
     )
   }
+  
+  func testIfconfigExpr31() {
+    AssertParse(
+      """
+      #if arch(x86_64)
+        debugPrint("x86_64")
+      1️⃣#else if arch(arm64)
+        debugPrint("arm64")
+      #else
+        debugPrint("Some other architecture.")
+      #endif
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          message: "unexpected 'if' keyword following '#else' conditional compilation directive; did you mean '#elseif'?",
+          fixIts: ["replace '#else if' by '#elseif'"]
+        ),
+      ]
```

Comment:
And one more request for the test. You can check that applying the Fix-It actually produces the expected source by  adding a `fixedSource:` parameter to `AssertParse`.

Reply:
Just did it. The `fixedSource` is being generated with a wrong indentation (it has a space before `#elseif`). Maybe this is happening because of the `locationMarker`?
```swift
      fixedSource: """
        #if arch(x86_64)
          debugPrint("x86_64")
         #elseif arch(arm64)
          debugPrint("arm64")
        #else
          debugPrint("Some other architecture.")
        #endif
        """
```

Reply:
Oh, good thing we checked it then. I think what’s happening is that the diagnostic generator is thinking that you want to replace the `if` by `#elseif` and is thus retaining the trailing space of the `if`. I haven’t tried it but you should be able to fix this by either setting `transferTrivia: false` in `makeMissing` or `leadingTrivia: []` in `makePresent`. The former would be the preferred solution.

Reply:
Thanks for the tip. 

`transferTrivia` alone didn't fix it, as it removed the newline in the source code. The empty array for `leadingTrivia` in `makePresent` wasn't enough either. Here's what made it work:
```swift
               [
                  .makeMissing(unexpectedBeforePoundKeyword, transferTrivia: false),
                  .makePresent(clause.poundKeyword, leadingTrivia: unexpectedBeforePoundKeyword.leadingTrivia)]
               ]
```